R/parse_income.R

```diff
@@ -13,109 +13,148 @@
 #' \code{"$"} by default, but could be e.g. \code{"AUD"} or \code{"USD"}.
 #' string "Negative income" is interpreted as \code{0}. If FALSE, \code{NA} is returned.
 #' @param .silent a boolean value. If FALSE (the default), the function will warn that  \code{NA}(s) were returned.
-#'
+#' @param .print If TRUE prints lookup table displaying inputted income range and the returned parsed value.
+#' 
 #' @return a numeric vector of incomes
 #'
 #' @importFrom stringr str_detect str_split str_replace_all
 #' @importFrom readr parse_number
 #'
 #' @export
 #'
-parse_income_range <- function(income_range,
-                               limit = "lower",
-                               max_income = Inf,
-                               is_zero = c("Nil income"),
-                               negative_as_zero = TRUE,
-                               dollar_prefix = "$",
-                               .silent = TRUE) {
-
-  # correct for use in regex
-  if (stringr::str_detect(dollar_prefix, "\\$")) {
-    dollar_prefix <- stringr::str_replace_all(dollar_prefix, "\\$", "\\\\$")
-  }
-
-  # run
-  purrr::map_dbl(income_range,
-                 .parse_income_range_single,
-                 .limit = limit,
-                 .negative_as_zero = negative_as_zero,
-                 .is_zero = is_zero,
-                 .max_income = max_income,
-                 .dollar_prefix = dollar_prefix,
-                 silent = .silent)
-
-}
-
-.parse_income_range_single <- function(x,
-                                .limit,
-                                .negative_as_zero,
-                                .is_zero,
-                                .max_income,
-                                .dollar_prefix,
-                                silent) {
+#'
 
```

Comment:
Sorry, the diff looks very strange. But essentially on the RHS, I added lines 27-34. Line 36 was where the original `parse_income_range` started and I changed the function name by adding an underscore.

Lastly I added lines 135-158 on the RHS to complete the join onto the lookup table and added the option to print. Everything else in the function is unchanged.

Reply:
Looks good!